vars/promotionHelper.groovy

```diff
@@ -33,17 +38,21 @@ void abortPreviousBuilds(RunWrapper promotedBuild) {
  */
 static int getUpstreamTriggerOrLatestBuildNumber(RunWrapper promotedBuild, String upstreamJobName) {
     int buildNum
+    def build
 
     // if build is trigger from upstream, retrieve the build number of the build that triggered this build
-    promotedBuild.getUpstreamBuilds().each { b ->
-        buildNum = b.number
-        println(">> Automatic trigger detected, using build ${b.getFullDisplayName()}")
+    def upstreamBuilds = promotedBuild.getUpstreamBuilds()
+    if (!upstreamBuilds.isEmpty()) {
+        build = upstreamBuilds.first()
+        println(">> Automatic trigger detected, using build ${build.getFullDisplayName()}")
+        buildNum = build.number
         return buildNum
     }
 
     // if build is trigger manually by a user, there is no upstream, so get the latest build number
-    def job = Jenkins.instance.getJob(upstreamJobName)
-    def build = job?.lastSuccessfulBuild
+    def instance = Jenkins.getInstanceOrNull()
+    def job = instance.getItemByFullName(upstreamJobName)
```

Comment:
Instance could be null at this point, due to having used the "getInstanceOrNull()" method. Using the "get()" method instead would make sure that there'd be a meaningful exception if the instance can't be retrieved. See https://javadoc.jenkins-ci.org/jenkins/model/Jenkins.html#getInstanceOrNull--

Reply:
Yeah reading the docs you're not supposed to use `getInstanceOrNull` except for specific cases. Updated to use `.get`.

I think the IDE suggested that since the `.instance` property was deprecated in later versions, so I went with the suggestion without RTFM.